src/platform/packages/private/kbn-grid-layout/grid/grid_panel/grid_panel.tsx

```diff
@@ -50,6 +50,14 @@ export const GridPanel = React.memo(({ panelId, rowId }: GridPanelProps) => {
     `;
   }, [gridLayoutStateManager, rowId, panelId]);
 
+  useEffect(() => {
+    return () => {
+      // remove reference on unmount
+      delete gridLayoutStateManager.panelRefs.current[rowId][panelId];
```

Comment:
We weren't cleaning up these references before - probably not a huge deal, but better to do this garbage collection  to prevent using unnecessary memory IMO